test/no-incorrect-deep-equal.js

```diff
@@ -40,6 +40,12 @@ ruleTester.run('no-incorrect-deep-equal', rule, {
 			test('x', t => {
 				t.notDeepEqual(expression, []);
 			});
+		`,
+		`
+			${header}
+			test('x', t => {
+				t.deepEqual(expression, /regex/);
```

Comment:
Also add a test with the arguments inversed: `t.deepEqual(/regex/, expression);`

Reply:
Ok I will do that